packages/api/src/external/carequality/document/process-document-query-results.ts

```diff
@@ -0,0 +1,167 @@
+import { makeFhirApi } from "../../fhir/api/api-factory";
+import { getAllPages } from "../../fhir/shared/paginated";
+import { isConvertible } from "../../fhir-converter/converter";
+import { MedicalDataSource } from "../../../external";
+import { appendDocQueryProgressWithSource } from "../../hie/append-doc-query-progress-with-source";
+
+const region = Config.getAWSRegion();
+const s3Utils = new S3Utils(region);
+const s3BucketName = Config.getMedicalDocumentsBucketName();
+
+export async function processDocumentQueryResults({
+  requestId,
+  patientId,
+  cxId,
+  documentQueryResults,
+}: {
+  requestId: string;
+  patientId: string;
+  cxId: string;
+  documentQueryResults: DocumentQueryResult[];
+}): Promise<void> {
+  const { log } = out(`CQ query docs - requestId ${requestId}, M patient ${patientId}`);
+
+  const docRefs = combineDocRefs(documentQueryResults);
+
+  try {
+    const docsToDownload = await getNonExistentDocRefs(docRefs, patientId, cxId);
+
+    const convertibleDocCount = docsToDownload.filter(doc =>
+      isConvertible(doc.contentType || undefined)
+    ).length;
+
+    log(`I have ${docsToDownload.length} docs to download (${convertibleDocCount} convertible)`);
+
+    await appendDocQueryProgressWithSource({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: {
+        status: "processing",
+        total: docsToDownload.length,
+      },
+      convertProgress: {
+        status: "processing",
+        total: convertibleDocCount,
+      },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    // TODO - INTRODUCED WHEN IMPLEMENTING CQ DOC RETRIEVAL
+    // NOTE MAKE SURE TO ADD DOC REFS TO TABLE WITH REQUESTID
+    // downloadDocs(docsToDownload, patientId, cxId, requestId);
+  } catch (error) {
+    const msg = `Failed to process documents in Carequality.`;
+    console.log(`${msg}. Error: ${errorToString(error)}`);
+
+    await appendDocQueryProgressWithSource({
+      patient: { id: patientId, cxId: cxId },
+      downloadProgress: { status: "failed" },
+      requestId,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+
+    capture.message(msg, {
+      extra: {
+        context: `cq.processingDocuments`,
+        error,
+        patientId: patientId,
+        requestId,
+        cxId,
+      },
+      level: "error",
+    });
+    throw error;
+  }
+}
+
+function combineDocRefs(documentQueryResults: DocumentQueryResult[]): DocumentReference[] {
+  return documentQueryResults.reduce((acc: DocumentReference[], curr) => {
+    const documentReferences = curr.data.documentReference ?? [];
+
+    return [...acc, ...documentReferences];
+  }, []);
+}
+
+const getNonExistentDocRefs = async (
+  documents: DocumentReference[],
+  patientId: string,
+  cxId: string
+): Promise<DocumentReference[]> => {
+  const [{ existingDocRefs, nonExistingDocRefs }, FHIRDocRefs] = await Promise.all([
+    filterOutExistingDocRefsS3(documents, patientId, cxId),
```

Comment:
just to confirm the desired behavior, I believe the Promise.all can never fail on the  `filterOutExistingDocRefsS3` clause here because `filterOutExistingDocRefsS3` uses Promise.allSettled. IF that isn't the desired behavior, then you could just change `filterOutExistingDocRefsS3` to use Promise.all

Reply:
This is similar to the flow we have in CW just refactored and it seems to work as expected